packages/core/chrome/core-chrome-browser/src/project_navigation.ts

```diff
@@ -71,13 +78,16 @@ export interface ChromeProjectNavigation {
    * The navigation tree representation of the side bar navigation.
    */
   navigationTree: ChromeProjectNavigationNode[];
+  /**
+   * The flattened navigation tree representation of the side bar navigation.
+   * This is optional, if not provided it will be computed from the navigationTree
+   */
+  navigationTreeFlattened?: Record<string, ChromeProjectNavigationNode>;
```

Comment:
Is this just for performance optimization or is there another reason? 

If this is only for performance to not flatten the same tree multiple times, then I think this is not worth it. This makes types/apis more complex (even if these are internal) and adds a bug surface where two trees can get out of sync.

If there is a performance concern, then I'd try to wrap flattenTree in memoize-one

Reply:
> Is this just for performance optimization or is there another reason?

If is for performance optimisation. The react component flattens the tree, then the project navigation service needs to flatten the tree too. I didn't think it was necessary to do it twice.

I still think it would be good to pass it if we have computed it already but maybe it is not worth it. I will see if there is a way to not make it public.

Reply:
See my suggestion about not passing it around, but wrapping the flattenTree with memoize-one. Maybe it is good enough

Reply:
I removed it as it was not needed with the debounce update.